File: Python_Patterns/Monitor.py
import threading
import time

class PhotoProcessor:
    def __init__(self):
        self._lock = threading.Lock()

    def process_photo(self, photo):
        with self._lock:
            print(f"Processing photo: {photo}")
            time.sleep(2)
            print(f"Photo processed: {photo}")

File: Lab_RGR/Monitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_RGR
{
    // Concurrency Pattern - Monitor
    public class PhotoProcessor
    {
        private object _lock = new object();

        public void ProcessPhoto(string photo)
        {
            lock (_lock)
            {
                Console.WriteLine($"Processing photo: {photo}");
                Thread.Sleep(2000);
                Console.WriteLine($"Photo processed: {photo}");
            }
        }
    }
}